source/WorkingPractices/Diagnostics/lfric_diagnostics.rst

```diff
@@ -1,11 +1,409 @@
+
+Several science schemes have dedicated modules to initialise and
+output diagnostics (steps 2 and 3 of the above list). See the modules
+in ``um_physics_interface/source/diagnostics/``.
+
+The following sections describe each of the steps in more detail.
+
+Creating a field record
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Diagnostics that are computed every time-step should normally be
+defined in the ``field_def_diags.xml`` file.
+
+.. topic:: Other field definition files
+
+   Other field definition files include the ``lfric_dictionary.xml``
+   file which lists prognostic fields (some of which may also be
+   output as diagnostic fields); the ``field_def_initial_diags.xml``
+   file holding definitions of diagnostics that optionally can be
+   output before the first time-step has run; and the
+   ``field_def_lbc_diags.xml`` file that supports reading and writing
+   of Lateral Boundary Conditions fields.
+
+A field definition looks like this:
+
+.. code-block:: xml
+
+  <field id="convection__shallow_dt" name="shallow_dt"
+  long_name="temperature_increment_from_shallow_convection" unit="K
+  s-1" grid_ref="full_level_face_grid" />
+
+The components of this definition are:
+
+   * The ``id`` string is used in the model code to identify the
+     diagnostic. The naming convention used by the LFRic atmosphere is
+     the section name followed by a double-understroke followed by a
+     descriptive name.
+   * The name and long name are only seen in the diagnostic
+     output. The names may be formally assigned such as by the CF
+     naming convention. In this case, the name is the same as the
+     suffix of the ID, but it is not always so.
+   * The units should be SI units. Again, these are only seen in the
+     diagnostic output file.
+   * The ``grid_ref`` attribute of this definition describes the
+     domain of the field. The example field above is represented in
+     the model as a :math:`\mathbb{W}_{theta}` field. Other field
+     types have different attributes as shown in the following table.
+
++-----------------------------------+----------------------------------------+
+|  Model field type                 |  Domain attributes                     |
++===================================+========================================+
+|  :math:`\mathbb{W}_{3}`           |  ``grid_ref="half_level_face_grid"``   |
++-----------------------------------+----------------------------------------+
+|  :math:`\mathbb{W}_{theta}`       |  ``grid_ref="full_level_face_grid"``   |
++-----------------------------------+----------------------------------------+
+|  :math:`\mathbb{W}_{2v}`          |  ``grid_ref="full_level_face_grid"``   |
++-----------------------------------+----------------------------------------+
+|  :math:`\mathbb{W}_{2h}`          |  ``grid_ref="half_level_edge_grid"``   |
++-----------------------------------+----------------------------------------+
+|  :math:`\mathbb{W}_{theta}`       |  ``grid_ref="full_level_face_grid"``   |
```

Comment:
Should there be multiple W_theta entries here?

Reply:
No. One removed.

Reply:
Think there were 3 before, now 2 L183 + L189?

Reply:
Eek! Now fixed and checked more carefully.

Reply:
Easily done, thanks.